src/redux/api-key/api-key-reducer.ts

```diff
@@ -0,0 +1,28 @@
+import { API_KEY_LOADED_EVENT, API_KEYS_REDUCER_KEY } from './constants';
+
+export type ApiKeyValue = {
+  name: string;
+  url: string;
+};
+
+type ApiKeyState = {
+  data: Array<Record<string, ApiKeyValue>>;
+};
+
+type Action = {
+  payload: Record<string, unknown>;
+  type: string;
```

Comment:
You can use restrict this type further to restrict to only the supported actions

Reply:
Restricted; currently only one supported action.